toutv/transport.py

```diff
@@ -73,15 +77,19 @@ def _do_query_url(self, url, params={}, timeout=20):
 
             return r
         except requests.exceptions.Timeout:
+            if retry_on_timeout and num_retries < 5:
```

Comment:
Take a look at how `_download_segment_with_retry` is implemented.  IMO it's clearer with the for loop that recursion.